portal-impl/src/com/liferay/portal/service/impl/UserLocalServiceImpl.java

```diff
@@ -3495,6 +3495,29 @@ public List<User> searchBySocial(
 			).build(),
 			start, end, orderByComparator);
 	}
+	
+	@Override
+	public List<User> searchBySocial(
+		long companyId, long[] groupIds, long[] userGroupIds, String keywords, int start, int end,
+		OrderByComparator<User> orderByComparator) {
+		
+		LinkedHashMap<String, Object> userParams =
+				LinkedHashMapBuilder.<String, Object>put(
+					"wildcardMode", WildcardMode.TRAILING
+				).build();
+		
+		if (!ArrayUtil.isEmpty(groupIds)) {
+			userParams.put("usersGroups", ArrayUtil.toLongArray(groupIds));
+		}
+		
+		if (!ArrayUtil.isEmpty(userGroupIds)) {
+			userParams.put("usersUserGroups", ArrayUtil.toLongArray(userGroupIds));
+		}
+		
```

Comment:
I think this can be simplified in something like this (please verify it)
```
return userFinder.findByKeywords(
	companyId, keywords, WorkflowConstants.STATUS_APPROVED,
	LinkedHashMapBuilder.<String, Object>put(
		"usersGroups", () -> {
			if (!ArrayUtil.isEmpty(groupIds)) {
				return ArrayUtil.toLongArray(groupIds);
			}
			
			return null;
	}).put(
		"usersUserGroups", () -> {
			if (!ArrayUtil.isEmpty(userGroupIds)) {
				return ArrayUtil.toLongArray(userGroupIds);
			}
			
			return null;
	}).put(
		"wildcardMode", WildcardMode.TRAILING
	).build(),
	start, end, orderByComparator);
```

Reply:
Verified. It worked as expected with this modification